lms/djangoapps/teams/static/teams/js/views/team_join.js

```diff
@@ -1,108 +1,115 @@
+                    _.bindAll(this, 'render', 'joinTeam', 'getUserTeamInfo');
+                    this.listenTo(this.model, "change", this.render);
+                },
 
-               render: function() {
-                   var message,
-                       showButton,
-                       teamHasSpace;
+                render: function() {
+                    var view = this,
+                        message,
+                        showButton,
+                        teamHasSpace;
+                    this.getUserTeamInfo(this.currentUsername, view.maxTeamSize).done(function (info) {
+                        teamHasSpace = info.teamHasSpace;
 
-                   var view = this;
-                   this.getUserTeamInfo(this.currentUsername, view.maxTeamSize).done(function (info) {
-                       teamHasSpace = info.teamHasSpace;
+                        // if user is the member of current team then we wouldn't show anything
+                        if (!info.memberOfCurrentTeam) {
+                            showButton = !info.alreadyMember && teamHasSpace;
 
-                       // if user is the member of current team then we wouldn't show anything
-                       if (!info.memberOfCurrentTeam) {
-                           showButton = !info.alreadyMember && teamHasSpace;
+                            if (info.alreadyMember) {
+                                message = info.memberOfCurrentTeam ? '' : view.alreadyMemberMessage;
+                            } else if (!teamHasSpace) {
+                                message = view.teamFullMessage;
+                            }
+                        }
 
-                           if (info.alreadyMember) {
-                               message = info.memberOfCurrentTeam ? '' : view.alreadyMemberMessage;
-                           } else if (!teamHasSpace) {
-                               message = view.teamFullMessage;
-                           }
-                       }
+                        view.$el.html(view.template({showButton: showButton, message: message}));
+                    });
+                    return view;
+                },
 
-                       view.$el.html(view.template({showButton: showButton, message: message}));
-                   });
-                   return view;
-               },
+                joinTeam: function () {
+                    var view = this;
+                    $.ajax({
+                        type: 'POST',
+                        url: view.teamMembershipsUrl,
+                        data: {'username': view.currentUsername, 'team_id': view.model.get('id')}
+                    }).done(function (data) {
+                        view.teamEvents.trigger('teams:update', {
+                            action: 'join',
+                            team: view.model
+                        });
+                        view.model.fetch()
```

Comment:
I'm wondering if the teamEvents trigger should happen in this "done" since the teams:update sends view.model as the team (and it won't be up-to-date until the fetch is complete).